esmvalcore/preprocessor/_regrid.py

```diff
@@ -705,29 +711,46 @@ def parse_vertical_scheme(scheme):
         The vertical interpolation scheme to use. Choose from
         'linear',
         'nearest',
-        'nearest_horizontal_extrapolate_vertical',
-        'linear_horizontal_extrapolate_vertical'.
+        'linear_extrapolate',
+        'nearest_extrapolate'.
 
     Returns
     -------
     (str, str)
         A tuple containing the interpolation and extrapolation scheme.
     """
+    # Issue warning when deprecated schemes are used
+    deprecated_schemes = {
+        'linear_horizontal_extrapolate_vertical': 'linear_extrapolate',
+        'nearest_horizontal_extrapolate_vertical': 'nearest_extrapolate',
+    }
+    if scheme in deprecated_schemes:
+        new_scheme = deprecated_schemes[scheme]
+        deprecation_msg = (
+            f"The vertical regridding scheme ``{scheme}`` has been deprecated "
+            f"in ESMValCore version 2.5 and is scheduled for removal in "
+            f"version 2.7. It has been replaced with the scheme "
+            f"``{new_scheme}``.")
```

Comment:
Perhaps make it clear that the new schemes are identical to the old ones with only a changed name?
Also, it might be worth leaving a link to the documentation?